src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,99 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const pagesArray = [];
+  const items = [];
+  const startIndex = (currentPage - 1) * perPage;
+  const lastIndex = Math.min(startIndex + perPage, total);
+
+  for (let i = 1; i <= Math.ceil(total / perPage); i++) {
+    pagesArray.push(i);
+  }
+
+  for (let i = startIndex; i < lastIndex; i++) {
+    items.push(i);
+  }
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames('page-item', { disabled: currentPage === 1 })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
```

Comment:
The `href` attribute is set to `#prev`, which can cause the page to jump to the top when clicked. Consider using `event.preventDefault()` in the `onClick` handler to prevent this default behavior.